File: backend/services/shrimp_type_service.go
package services
import (
	"errors"
	"cronos-backend/models"
	"cronos-backend/repositories"
	"github.com/google/uuid"
)
type ShrimpTypeService struct {
	ShrimpTypeRepo *repositories.ShrimpTypeRepository
}
func NewShrimpTypeService(repo *repositories.ShrimpTypeRepository) *ShrimpTypeService {
	return &ShrimpTypeService{ShrimpTypeRepo: repo}
}
func (s *ShrimpTypeService) Create(st *models.ShrimpType) error {
	return s.ShrimpTypeRepo.Create(st)
}
func (s *ShrimpTypeService) GetAll() ([]models.ShrimpType, error) {
	return s.ShrimpTypeRepo.FindAll()
}
func (s *ShrimpTypeService) Update(id uuid.UUID, req *models.ShrimpType) (*models.ShrimpType, error) {
	st, err := s.ShrimpTypeRepo.FindByID(id)
	if err != nil {
		return nil, errors.New("shrimp type not found")
	}
	if req.Name != "" {
		st.Name = req.Name
	}
	if req.Description != "" {
		st.Description = req.Description
	}
	if req.Image != "" {
		st.Image = req.Image
	}
	if err := s.ShrimpTypeRepo.Update(st); err != nil {
		return nil, errors.New("failed to update")
	}
	return st, nil
}
func (s *ShrimpTypeService) Delete(id uuid.UUID) error {
	return s.ShrimpTypeRepo.Delete(id)
}

File: backend-spring/src/main/java/com/cronos/service/ShrimpTypeService.java
package com.cronos.service;

import com.cronos.entity.ShrimpType;
import com.cronos.repository.ShrimpTypeRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class ShrimpTypeService {
    private final ShrimpTypeRepository repo;

    public ShrimpType create(ShrimpType st) { repo.save(st); return st; }
    public List<ShrimpType> getAll() { return repo.findAllByOrderByNameAsc(); }
    public ShrimpType update(UUID id, ShrimpType req) {
        ShrimpType st = repo.findById(id).orElseThrow(() -> new RuntimeException("shrimp type not found"));
        if (req.getName() != null && !req.getName().isEmpty()) st.setName(req.getName());
        if (req.getDescription() != null && !req.getDescription().isEmpty()) st.setDescription(req.getDescription());
        if (req.getImage() != null && !req.getImage().isEmpty()) st.setImage(req.getImage());
        repo.save(st); return st;
    }
    public void delete(UUID id) { repo.deleteById(id); }
}
